konrad/cloud.py

```diff
@@ -287,147 +241,200 @@ def __init__(self, z, cloud_fraction,
+            z (ndarray): height array, on which the interpolation is performed
+        Returns:
+            scipy.interpolate.interpolate.interp1d
         """
-        cloud_fraction_array = np.zeros(z.shape)
-        cloud_fraction_array[(z < cloud_top) & (
-                z > cloud_top-depth)] = cloud_fraction_in_cloud
-        cloud_fraction = DataArray(cloud_fraction_array, dims=('mid_levels',),
-                                   attrs={'units': 'dimensionless'})
-
-        mass_ice = self.calculate_mass_cloud(cloud_fraction_array, dz_from_z(z),
-                                             ice_density)
+        normed_height = z - self._norm_level
 
-        super().__init__(
-            z,
-            cloud_fraction=cloud_fraction,
-            mass_ice=mass_ice,
-            ice_particle_size=ice_particle_size
+        interpolation_f = interp1d(
+            normed_height,
+            cloud_parameter.values,
+            fill_value='extrapolate',
+            axis=0,
         )
+        return interpolation_f
 
-        self._ice_density = ice_density
-        self._norm_level = cloud_top
-        self._f = self.interpolation_function(
-            self.cloud_area_fraction_in_atmosphere_layer, z)
+    def shift_property(self, cloud_parameter, interpolation_f, z, norm_new):
+        """Shift the cloud area fraction according to a normalisation level.
+        Maintain the thickness of the cloud [m].
 
-    def update_cloud_profile(self, atmosphere, convection, **kwargs):
-        """ Keep the cloud attached to the convective top.
+        Parameters:
+            cloud_parameter (DataArray): cloud property to be shifted
+            interpolation_f (scipy.interpolate.interpolate.interp1d):
+                interpolation object calculated by interpolation_function
+            z (ndarray): height array [m]
+            norm_new (int / float): normalisation height [m]
+
+        Returns:
+            DataArray: shifted cloud property
         """
-        norm_new = convection.get('convective_top_height')[0]
-        z = atmosphere.get('z', keepdims=False)
-        dz = dz_from_z(z)
+        if norm_new is not np.nan:
+            # Move the cloud to the new normalisation level, if there is one.
+            cloud_parameter.values = interpolation_f(z - norm_new)
+        else:
+            # Otherwise keep the cloud where it is.
+            cloud_parameter.values = interpolation_f(z - self._norm_level)
+
+        return cloud_parameter
+
+    def shift_cloud_profile(self, plev, norm_new):
+        if self._norm_level is None:
+            self._norm_level = norm_new
+
+            self._interp_cldf = self.interpolation_function(
+                cloud_parameter=self.cloud_area_fraction_in_atmosphere_layer,
+                z=plev,
+            )
+
+            self._interp_sw = self.interpolation_function(
+                cloud_parameter=self.shortwave_optical_thickness_due_to_cloud,
+                z=plev,
+            )
+
```

Comment:
We need to be a bit careful here. I think I'd rather do the interpolation on model level indices, ie `z = range(0, plev.size)`. Then the normalisation level would also need to be in terms of model level indices. I can make the edits for this if you like.

Reply:
Agreed! I will finish our _The winner takes it all_ approach for adding two clouds and push it 😉